staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/lastappliedupdater.go

```diff
@@ -115,3 +120,14 @@ func buildLastApplied(obj runtime.Object) (string, error) {
 	}
 	return string(lastApplied), nil
 }
+
+func isAnnotationsValid(annotations map[string]string) error {
```

Comment:
this is fine for this PR, since we'll want to pick this back to 1.19-1.21, but in a follow-up, we should make the apimachinery annotations size validation public and use that instead of a fork so the logic doesn't drift

can you open an issue to track that for 1.22?

Reply:
Will do. Another option that sounds similar is we could call ValidateAnnotations and check that the error list is empty, but might be not ideal if ValidateAnnotations expands too much

Reply:
yeah, I don't want false-positives if another annotation in the update had a key with a disallowed character or something

Reply:
Done: https://github.com/kubernetes/kubernetes/issues/102110